From 8451685fd4b74252f3f7cf203985b1306e00f1c9 Mon Sep 17 00:00:00 2001
From: Tom Lukaszewicz <tluk@chromium.org>
Date: Thu, 14 Jul 2022 01:57:51 +0000
Subject: [PATCH 084/110] [sheriff] Disable flaky
 WebContentsObserverBrowserTests on Mac

(cherry picked from commit 95fe4b6b97c1a1d5006b9d496695e1be87d40daa)

Bug: 1288573
Change-Id: I09511c3c4fdce6a508db17b67c07de2a4f205e8e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3598093
Owners-Override: Tom Lukaszewicz <tluk@google.com>
Reviewed-by: Tom Lukaszewicz <tluk@google.com>
Commit-Queue: Tom Lukaszewicz <tluk@google.com>
Cr-Original-Commit-Position: refs/heads/main@{#994538}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3760779
Reviewed-by: Thomas Lukaszewicz <tluk@chromium.org>
Commit-Queue: Joe Downing <joedow@chromium.org>
Owners-Override: Joe Downing <joedow@chromium.org>
Cr-Commit-Position: refs/branch-heads/5005@{#1243}
Cr-Branched-From: 5b4d9450fee01f821b6400e947b3839727643a71-refs/heads/main@{#992738}
---
 .../web_contents_observer_browsertest.cc      | 26 ++++++++++++++++---
 1 file changed, 23 insertions(+), 3 deletions(-)

diff --git a/content/browser/web_contents/web_contents_observer_browsertest.cc b/content/browser/web_contents/web_contents_observer_browsertest.cc
index 3922bb02ca1f..e38227fafe84 100644
--- a/content/browser/web_contents/web_contents_observer_browsertest.cc
+++ b/content/browser/web_contents/web_contents_observer_browsertest.cc
@@ -375,8 +375,14 @@ using CookieAccess = CookieTracker::CookieAccessDescription;
 
 }  // namespace
 
+// TODO(https://crbug.com/1288573): Flaky on Mac.
+#if BUILDFLAG(IS_MAC)
+#define MAYBE_CookieCallbacks_MainFrame DISABLED_CookieCallbacks_MainFrame
+#else
+#define MAYBE_CookieCallbacks_MainFrame CookieCallbacks_MainFrame
+#endif
 IN_PROC_BROWSER_TEST_F(WebContentsObserverBrowserTest,
-                       CookieCallbacks_MainFrame) {
+                       MAYBE_CookieCallbacks_MainFrame) {
   CookieTracker cookie_tracker(web_contents());
 
   GURL first_party_url("http://a.com/");
@@ -420,8 +426,16 @@ IN_PROC_BROWSER_TEST_F(WebContentsObserverBrowserTest,
   cookie_tracker.cookie_accesses().clear();
 }
 
+// TODO(https://crbug.com/1288573): Flaky on Mac.
+#if BUILDFLAG(IS_MAC)
+#define MAYBE_CookieCallbacks_MainFrameRedirect \
+  DISABLED_CookieCallbacks_MainFrameRedirect
+#else
+#define MAYBE_CookieCallbacks_MainFrameRedirect \
+  CookieCallbacks_MainFrameRedirect
+#endif
 IN_PROC_BROWSER_TEST_F(WebContentsObserverBrowserTest,
-                       CookieCallbacks_MainFrameRedirect) {
+                       MAYBE_CookieCallbacks_MainFrameRedirect) {
   CookieTracker cookie_tracker(web_contents());
 
   GURL first_party_url("http://a.com/");
@@ -541,8 +555,14 @@ IN_PROC_BROWSER_TEST_F(WebContentsObserverBrowserTest,
   cookie_tracker.cookie_accesses().clear();
 }
 
+// TODO(https://crbug.com/1288573): Flaky on Mac.
+#if BUILDFLAG(IS_MAC)
+#define MAYBE_CookieCallbacks_Subresource DISABLED_CookieCallbacks_Subresource
+#else
+#define MAYBE_CookieCallbacks_Subresource CookieCallbacks_Subresource
+#endif
 IN_PROC_BROWSER_TEST_F(WebContentsObserverBrowserTest,
-                       CookieCallbacks_Subresource) {
+                       MAYBE_CookieCallbacks_Subresource) {
   CookieTracker cookie_tracker(web_contents());
 
   GURL first_party_url("http://a.com/");
-- 
2.38.1

